src/viam/sdk/module/service.cpp

```diff
@@ -250,7 +250,14 @@ void ModuleService::serve() {
 }
 
 ModuleService::~ModuleService() {
-    // TODO(RSDK-5509): Run registered cleanup functions here.
+    try {
+        cleanup_function_();
+    } catch (const std::exception& xcp) {
+        BOOST_LOG_TRIVIAL(error) << "Module cleanup function failed with exception " << xcp.what();
+    } catch (...) {
+        BOOST_LOG_TRIVIAL(error) << "Module cleanup function failed with unknown exception";
+    }
```

Comment:
I think this will cause error logging in cases where the cleanup function is never set (it certainly does in godbolt). We should maybe make this be a pointer to a func, and then only try to call when the funcptr is not null?

Reply:
Ooh you are very right; let me switch to a pointer and also do some more testing for these error logging paths.